xgcm/grid.py

```diff
@@ -1150,6 +1179,16 @@ def __init__(
                     "mapping axis name to a boundary option; a number or None."
                 )
 
+            if isinstance(direction, dict):
```

Comment:
We could add some logic to infer the values of direction using cf-xarray, and then overwrite values from there with the ones manually provided.

Reply:
We only need to look for the `positive` attribute.
``` python
if ....attrs["positive"] == "up":
	axis_direction = "increasing"
elif ....attrs["positive"] == "down":
    axis_direction = "decreasing"
else:
	raise ValueError
```